lib/grizzly/zwave/command_classes/barrier_operator.ex

```diff
@@ -0,0 +1,108 @@
+defmodule Grizzly.ZWave.CommandClasses.BarrierOperator do
+  @moduledoc """
+  "BarrierOperator" Command Class
+
+  The Barrier Operator Command Class is used to control and query the status of motorized barriers.
+  """
+
+  @behaviour Grizzly.ZWave.CommandClass
+
+  alias Grizzly.ZWave.DecodeError
+
+  use Bitwise
+
+  @type target_value :: :open | :close
+  @type state :: :closed | 0x01..0x63 | :closing | :stopped | :opening | :open
+  @type subsystem_type :: :audible_notification | :visual_notification
+  @type subsystem_state :: :on | :off
+
+  @impl Grizzly.ZWave.CommandClass
+  def byte(), do: 0x66
+
+  @impl Grizzly.ZWave.CommandClass
+  def name(), do: :barrier_operator
+
+  @spec target_value_to_byte(:close | :open) :: 0x00 | 0xFF
+  def target_value_to_byte(:close), do: 0x00
+  def target_value_to_byte(:open), do: 0xFF
+
+  @spec target_value_from_byte(byte) ::
+          {:error, Grizzly.ZWave.DecodeError.t()} | {:ok, :close | :open}
+  def target_value_from_byte(0x00), do: {:ok, :close}
+  def target_value_from_byte(0xFF), do: {:ok, :open}
+  def target_value_from_byte(byte), do: {:error, %DecodeError{value: byte, param: :target_value}}
+
+  @spec state_to_byte(:closed | :closing | :open | :opening | :stopped | 0x01..0x63) :: byte
```

Comment:
Can we use the `state()` type that you defined above in this spec?

Reply:
Done